### ISSUE TITLE
jade extended / included files not refreshing "if" logic

### ISSUE THREAD

**Issue by chisleu** [pugjs/pug#1763]
I provide a user object to a jade view which indicates if there is a user or not.
I can stop and restart node or nodemon and I stay logged in with my cookie, as expected.
Once I login, jade refreshes to display the user, as expected.

However, when I attempt to logout, and unset my session variable (and save), jade doesn't seem to respond. Here is a console log:

```
POST /login/bye 302 16.859 ms - 58
{ cookie:
   { path: '/',
     _expires: Tue Dec 09 2014 06:08:44 GMT-0500 (EST),
     originalMaxAge: 604782193,
     httpOnly: true } }
No user...
index GET
index no user
GET / 200 74.732 ms - 2149
```

You can see that I call /login/bye, then my login middleware shows that the cookie no longer has a user in it, and the if responds "No user...". Then this code is run in my index.js route:

```
router.get('/',
    function (req, res) {
        console.log('index GET');
        if (!req.session.user) {
            console.log('index no user');
            return res.render('index', {title: 'A TEST Experiment'});
        }

        return res.render('index', {
            user: req.user,
            title: 'A TEST Experiment'
        });
    }
);
```

Notice that this code indicates that the session is not there, so index.jade is not passed a "user" object. However, it doesn't seem to respond to this lack of a user inside the jade, and displays the values that used to exist, but have been deleted.

Here is the code to /login/bye for reference:

```
// Logout
router.all('/bye', function (req, res) {
    // Don't question it... logout.
    delete req.user;
    delete req.session.user;
    req.session.save(function(err) {
        if (err) console.log('ERROR: logout failed');
        return res.redirect('/');
    });
});
```

Even the req.user is deleted at this point, but I think this is irrelevant as the new GET / is a new connection.

TLDR:
Jade seems to cache variables passed into it beyond their scope, or express is being very naughty. I'm not sure which.

NOTE:
- It just occurred to me to explicitly change user to false, and that DOES fix the problem. However shouldn't this be unneeded?
- I updated to 1.8.1 and confirmed this was still an issue (at least for me)

**Comment by seudein** [comment]
having this same issue. Was fine before, but now the extended page won't refresh (but base page does).
